server/base/src/main/java/org/apache/accumulo/server/metadata/ServerAmpleImpl.java

```diff
@@ -147,6 +154,63 @@ public void putGcFileAndDirCandidates(TableId tableId, Collection<ReferenceFile>
     }
   }
 
+  @Override
+  public void addBulkLoadInProgressFlag(String path, long fateTxid) {
+
+    // Bulk Import operations are not supported on the metadata table, so no entries will ever be
+    // required on the root table.
+    KeyExtent extent = new KeyExtent(TableId.of("placeholder"), null, null);
+    Mutation m = new Mutation(BlipSection.getRowPrefix() + path);
+    m.put(EMPTY_TEXT, EMPTY_TEXT, new Value(FateTxId.formatTid(fateTxid)));
+
+    // Uses a "placeholder" table name to create a metadata table writer.
+    try (BatchWriter bw = createWriter(extent.tableId())) {
+      bw.addMutation(m);
+    } catch (MutationsRejectedException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void removeBulkLoadInProgressFlag(String path) {
+
+    // Bulk Import operations are not supported on the metadata table, so no entries will ever be
+    // required on the root table.
+    KeyExtent extent = new KeyExtent(TableId.of("placeholder"), null, null);
+    Mutation m = new Mutation(BlipSection.getRowPrefix() + path);
+    m.putDelete(EMPTY_TEXT, EMPTY_TEXT);
+
+    // Uses a "placeholder" table name to create a metadata table writer.
+    try (BatchWriter writer = createWriter(extent.tableId())) {
```

Comment:
Could directly create the batch writer and drop the extent.


```suggestion
    Mutation m = new Mutation(BlipSection.getRowPrefix() + path);
    m.putDelete(EMPTY_TEXT, EMPTY_TEXT);

    try (BatchWriter writer =  context.createBatchWriter(MetadataTable.NAME)) {
```

Reply:
I agree that doing that reduces the overall code and cleans up the logic. 

Should I also remove `ServerAmpleImpl.createWriter()`? 

Outside of my uses, it's only used by two methods that already compare the tableID against `root` before calling `createWriter()`. They could easily be refactored to call the correct context directly instead.

Reply:
> Should I also remove ServerAmpleImpl.createWriter()?

I think it could be removed and replaced with : `context.createBatchWriter(DataLevel.of(tableId).metaTable())`